apps/linear_algebra/src/blas_l3_generators.cpp

```diff
@@ -0,0 +1,110 @@
+#include <vector>
+#include "Halide.h"
+
+using namespace Halide;
+
+namespace {
+
+// Generator class for BLAS gemm operations.
+template<class T>
+class GEMMGenerator :
+        public Generator<GEMMGenerator<T>> {
+  public:
+    typedef Generator<GEMMGenerator<T>> Base;
+    using Base::target;
+    using Base::get_target;
+    using Base::natural_vector_size;
+
+    GeneratorParam<bool> assertions_enabled_ = {"assertions_enabled", false};
+    GeneratorParam<bool> use_fma_ = {"use_fma", false};
+    GeneratorParam<bool> vectorize_ = {"vectorize", true};
+    GeneratorParam<bool> parallel_ = {"parallel", true};
+    GeneratorParam<int>  block_size_ = {"block_size", 1 << 5};
+    GeneratorParam<bool> transpose_A_ = {"transpose_A", false};
+    GeneratorParam<bool> transpose_B_ = {"transpose_B", false};
+
+    // Standard ordering of parameters in GEMM functions.
+    Param<T>   a_ = {"a", 1.0};
+    ImageParam A_ = {type_of<T>(), 2, "A"};
+    ImageParam B_ = {type_of<T>(), 2, "B"};
+    Param<T>   b_ = {"b", 1.0};
+    ImageParam C_ = {type_of<T>(), 2, "C"};
+
+    void SetupTarget() {
+        if (!assertions_enabled_) {
+            target.set(get_target()
```

Comment:
Does this have a performance impact? I could imagine it mattering for very small matrices. Still, nice to have it here for testing whether it matters.